server/src/main/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapper.java

```diff
@@ -0,0 +1,596 @@
+                throw new MapperParsingException("failed to parse field [{}] of type [{}]", e, fieldType().name(),
+                    fieldType().typeName());
+            }
+            context.addIgnoredField(fieldType.name());
+        }
+    }
+
+    private void indexShape(ParseContext context, Shape shape) {
+        List<IndexableField> fields = new ArrayList<>(Arrays.asList(fieldType().defaultPrefixTreeStrategy().createIndexableFields(shape)));
+        createFieldNamesField(context, fields);
+        for (IndexableField field : fields) {
+            context.doc().add(field);
+        }
+    }
+
+    @Override
+    protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
+        super.doXContentBody(builder, includeDefaults, params);
+
+        if (includeDefaults || fieldType().tree().equals(DeprecatedParameters.Defaults.TREE) == false) {
+            builder.field(DeprecatedParameters.Names.TREE.getPreferredName(), fieldType().tree());
+        }
+
+        if (fieldType().treeLevels() != 0) {
+            builder.field(DeprecatedParameters.Names.TREE_LEVELS.getPreferredName(), fieldType().treeLevels());
+        } else if(includeDefaults && fieldType().precisionInMeters() == -1) { // defaults only make sense if precision is not specified
+            if (DeprecatedParameters.PrefixTrees.GEOHASH.equals(fieldType().tree())) {
+                builder.field(DeprecatedParameters.Names.TREE_LEVELS.getPreferredName(),
+                    DeprecatedParameters.Defaults.GEOHASH_TREE_LEVELS);
+            } else if (DeprecatedParameters.PrefixTrees.LEGACY_QUADTREE.equals(fieldType().tree())) {
+                builder.field(DeprecatedParameters.Names.TREE_LEVELS.getPreferredName(),
+                    DeprecatedParameters.Defaults.QUADTREE_LEVELS);
+            } else if (DeprecatedParameters.PrefixTrees.QUADTREE.equals(fieldType().tree())) {
+                builder.field(DeprecatedParameters.Names.TREE_LEVELS.getPreferredName(),
+                    DeprecatedParameters.Defaults.QUADTREE_LEVELS);
+            } else {
+                throw new IllegalArgumentException("Unknown prefix tree type [" + fieldType().tree() + "]");
+            }
+        }
+        if (fieldType().precisionInMeters() != -1) {
+            builder.field(DeprecatedParameters.Names.PRECISION.getPreferredName(),
+                DistanceUnit.METERS.toString(fieldType().precisionInMeters()));
+        } else if (includeDefaults && fieldType().treeLevels() == 0) { // defaults only make sense if tree levels are not specified
+            builder.field(DeprecatedParameters.Names.PRECISION.getPreferredName(),
+                DistanceUnit.METERS.toString(50));
+        }
+        if (includeDefaults || fieldType().strategy() != DeprecatedParameters.Defaults.STRATEGY) {
+            builder.field(DeprecatedParameters.Names.STRATEGY.getPreferredName(), fieldType().strategy().getStrategyName());
```

Comment:
let's always have the strategy in the xcontent for legacy shapes to avoid any confusion? ie. remove the if statement and always execute the body

Reply:
:+1:  great idea